cppForSwig/BlockDataMap.cpp

```diff
@@ -70,7 +79,32 @@ void BlockData::deserialize(const uint8_t* data, size_t size,
          offsetOuts[y],
          offsetOuts[y + 1] - offsetOuts[y]));
 
-      tx->lockTime_ = READ_UINT32_LE(brr.getCurrPtr() + offsetOuts.back());
+      // Add all the old serialization data to noWitData_
+      tx->noWitData_.append(brr.getCurrPtr(), 4);
+      uint32_t txInOutSize = offsetOuts.back() - offsetIns[0];
+      uint32_t viInLen;
+      if(useWitnesses) {
+         BtcUtils::readVarInt(brr.getCurrPtr() + 6, &viInLen);
+         tx->noWitData_.append(brr.getCurrPtr() + 4 + 2, txInOutSize + viInLen);
```

Comment:
do not produce the noWitData at deser. 

first it creates a copy with no guarantee it will used. its only purpose is to get the txhash, and during scans, maybe 1000tx out of the entire blockchain will need to be hashed (the tx that have outputs relevant to our wallet, to track utxos),

next, it eats a lot more ram because of maintaining a copy vs a pointer to the data on the disk mmap. tx stripped of their witness data should be created on the fly, whenever the txhash is asked for. it avoids all copies for legacy tx, only performs copies when needed for segwit tx, and dumps the copied data after hash is computed.

this is very speed sensitive code, avoid copies whenever possible.